File: Curso_avanzado_APIs/Laravel_app/Models/Alumno.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Alumno extends Model
{
    protected $table = 'alumnos';
    public $timestamps = false;
    protected $fillable = [
        'nombre',
        'edad',
        'direccion',
        'correo'
    ];
    public function inscripciones(): HasMany
    {
        return $this->hasMany(AlumnoGrupo::class, 'id_alumno');
    }
}

File: Curso_avanzado_APIs/FastAPI_app/app/models/alumnos/alumno.py
# app/models/alumnos/alumno.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from app.config.base import Base

class AlumnoModel(Base):
    __tablename__ = "alumnos"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    nombre = Column(String(50), nullable=False)
    edad = Column(Integer, nullable=True)
    direccion = Column(String(100), nullable=True)
    correo = Column(String(50), nullable=False, unique=True)

    # Relación muchos a muchos con tabla intermedia
    inscripciones = relationship(
        "AlumnoGrupoModel",
        back_populates="alumno"
    )
